cnd/src/db/created_swap.rs

```diff
@@ -0,0 +1,237 @@
+        let role = created.role;
+
+        let swap = InsertableSwap::new(created.swap_id, created.peer.clone(), created.role);
+        self.save_swap(&swap).await?;
+
+        if let Some(address_hint) = created.address_hint {
+            self.save_address_hint(created.peer, &address_hint).await?;
+        }
+
+        // Save the herc20 details.
+        let redeem_identity = match role {
+            Role::Alice => None,
+            Role::Bob => Some(Text(EthereumAddress::from(created.alpha.identity))),
+        };
+        let refund_identity = match role {
+            Role::Alice => Some(Text(EthereumAddress::from(created.alpha.identity))),
+            Role::Bob => None,
+        };
+        let herc = InsertableHerc20 {
+            swap_id: 0, // FK, set during save.
+            amount: Text(created.alpha.amount.into()),
+            chain_id: U32(created.alpha.chain_id),
+            expiry: U32(created.alpha.absolute_expiry),
+            hash_function: Text(HashFunction::Sha256),
+            token_contract: Text(created.alpha.token_contract.into()),
+            redeem_identity,
+            refund_identity,
+            ledger: "alpha".to_string(),
+        };
+
+        self.save_herc20(local_swap_id, &herc).await?;
+
+        // Save the halight details.
+        let redeem_identity = match role {
+            Role::Alice => Some(Text(created.beta.identity)),
+            Role::Bob => None,
+        };
+        let refund_identity = match role {
+            Role::Alice => None,
+            Role::Bob => Some(Text(created.beta.identity)),
+        };
+        let network = BitcoinNetwork::from_str(&created.beta.network)?;
+        let halight = InsertableHalight {
+            swap_id: 0, // FK, set during save.
+            amount: Text(created.beta.amount.into()),
+            network: Text(network),
+            chain: "bitcoin".to_string(), // FIXME: What should this be?
+            cltv_expiry: U32(created.beta.cltv_expiry),
+            hash_function: Text(HashFunction::Sha256),
+            redeem_identity,
+            refund_identity,
+            ledger: "beta".to_string(),
+        };
+        self.save_halight(local_swap_id, &halight).await?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Load<CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>> for Sqlite {
+    async fn load(
+        &self,
+        swap_id: LocalSwapId,
+    ) -> anyhow::Result<Option<CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>>> {
+        let swap = self.load_swap(swap_id).await?;
+        let herc20 = self.load_herc20(swap_id).await?;
+        let halight = self.load_halight(swap_id).await?;
+
+        let role = swap.role.0;
+        let peer = swap.counterparty_peer_id.0;
+
+        let address_hint = self.load_address_hint(&peer).await.ok();
+
+        let alpha_identity = match role {
+            Role::Alice => herc20
+                .refund_identity
+                .ok_or_else(|| Error::IdentityNotSet)?,
```

Comment:
Follow up ticket: https://github.com/comit-network/comit-rs/issues/2614

@tcharding @D4nte  I'd suggest we actually do this ticket "on the fly" with the following tickets to come, because we will understand better what we need once we actually start tackling the issues. I'd pull this ticket into the sprint.

Reply:
mmm, whats the action for this comment?  Remove Load<InProgressSwap> altogether from this PR and leave it for another time.  Or add two different load API methods for these use cases?

Reply:
> mmm, whats the action for this comment?

Leave things as is and improve iteratively as we use the APIs. I think it will become obvious once the APIs are in use.
You can already see it in the test and in the previous PR that still contained code for the HTTP routes, one could see it as-well :)